more_itertools/more.py

```diff
@@ -424,3 +424,61 @@ def windowed(seq, n, fillvalue=None):
     for item in it:
         append(item)
         yield tuple(window)
+
+
+def separate(iterable, keys=None, fn=None):
```

Comment:
(Repeating this because it got hidden by the partition→separate rename.) Is there any reason we can't determine the keys dynamically and thus obviate the need for an arg? Then we could rename `fn` to `key` and look like `sort()` et al. Obviously, we'd have to return a custom object that implements `__getitem__` rather than a dict....